ckzg/build.sh

```diff
@@ -22,7 +22,7 @@ rm -rf c-kzg
 print_msg "Cloning 4844"
 git clone https://github.com/ethereum/c-kzg-4844.git
 cd c-kzg-4844 || exit 1
-git -c advice.detachedHead=false checkout $C_KZG_4844_GIT_HASH
+git -c advice.detachedHead=false checkout "$C_KZG_4844_GIT_HASH"
```

Comment:
Removed ckzg from the workspace members with [8630e29](https://github.com/sifraitech/rust-kzg/commit/8630e29ec6aa599e99af472aa9d24c66a66b7022) because c-kzg-4844 already does what ckzg was intended to do. Moreover, the c-kzg-4844 bindings enable us to link our highly parallelized functions from the blst backend to those of c-kzg-4844. In the future, the ckzg project may become relevant if we decide to implement Nvidia CUDA or SIMD parallelizations on the C code.

Regarding those errors when you run clippy, it shouldn't build static libraries to do syntax and semantic analysis on the code (i.e. plain text), so it's either clippy doing something it shouldn't do or we do something wrong.

Reply:
It is in the build script, so Clippy will build it when invoked on a crate. Clippy is basically `cargo check` + additional lints and `cargo check` is essentially `cargo build` without linking step. So build.rs is invoked for all since it might be used to generate Rust code that will be later compiled, it will not skip that without explicit instructions.